### ISSUE TITLE
[indent] False positive with ternary statements inside template literals inside objects

### ISSUE THREAD

**Issue by almeidx** [typescript-eslint/typescript-eslint#2800]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```ts
const o = {
  a: {
    b: {
      name: () => `
        ${abc
          ? ''
          : ''}
      `,
    },
  },
};
```

```JSON
{
  "rules": {
    "@typescript-eslint/indent": [
      "error",
      2,
      {
        "ArrayExpression": 1,
        "CallExpression": {
          "arguments": 1
        },
        "flatTernaryExpressions": false,
        "FunctionDeclaration": {
          "body": 1,
          "parameters": 1
        },
        "FunctionExpression": {
          "body": 1,
          "parameters": 1
        },
        "ignoreComments": false,
        "ImportDeclaration": 1,
        "ObjectExpression": 1,
        "outerIIFEBody": 1,
        "SwitchCase": 1,
        "VariableDeclarator": 1
      }
    ]
  }
}
```

**Expected Result**

Nothing. The code is okay.

**Actual Result**

ESLint says line 6 and 7 should have an indentation of 4 instead of 10

**Additional Info**

*This might be a rule misconfiguration on my end, and if that's the case, my apologies*

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.8.1` |
| `@typescript-eslint/parser`        | `4.8.1` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `14.15.0` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1824

---

Also it looks like there's no TS specific code in your example, so its likely a problem in the base rule. Consider raising a bug against core eslint.
